File: chap10/run-model10-5.R
library(cmdstanr)

d <- read.csv('input/data-mix1.csv')
data <- list(N=nrow(d), Y=d$Y)
init <- list(a=0.5, mu=c(0,5), sigma=c(1,1))

model <- cmdstan_model(stan_file='model/model10-5.stan')
fit <- model$sample(data=data, init=function() init, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model10-5.RDS')

File: chap10/run-model10-5.py
import pandas
import cmdstanpy

d = pandas.read_csv('input/data-mix1.csv')
data = {'N':len(d), 'Y':d.Y}

def init_fun():
    return {'a':0.5, 'mu':[0,5], 'sigma':[1,1]}

model = cmdstanpy.CmdStanModel(stan_file='model/model10-5.stan')
fit = model.sample(data=data, seed=123, inits=init_fun(), parallel_chains=4)
fit.save_csvfiles('output/result-model10-5')
